### DISCUSSION TITLE
Are there plans for fleet-agnostic problem-sets?

### DISCUSSION THREAD

**Discussion by cnpryer** [reinterpretcat/vrp#43] [Q&A] [answered]
Hi! Love this project. If you're accepting PRs I'd love to help out if I stumble on ways to, especially if it's related to the content below.

I work for a 3PL so we often deal with problems where _fleets_ are, in a way, out-sourced to brokerage markets or the vehicle assignment is more flexible to carrier-relations we may have already existing in our networks. This has me playing around with different solvers and libraries to test out heuristic configuration that favors less-strict fleet data.

One way I'll typically approach this is to set up a fleet with vehicles existing at each source node, but the vehicle wouldn't have any particular end-node configured.

I plan to spend more time with this project to see if maybe I'm missing something, but in the mean-time is this type of problem compatible with the goals of this project?

Bonus: Another pain-point with solvers in the 3PL world is often cost-definition flexibility. LTL shipping utilizes a weight-break tier system for definition costs associated with using those _assets_. I'd be curious of ways to introduce that into a project like this if it isn't already possible.

**Comment by cnpryer** [comment, marked as answer]
Maybe answering my own question with this https://github.com/reinterpretcat/vrp/discussions/26#discussioncomment-1332994

**Reply by reinterpretcat** [reply]
Hi, thanks for feedback!

Recently, I've added limited support for tplib95 format which can be used to benchmark some Capacitated VRP variants. Also this benchmark is supported too: https://www.sintef.no/projectweb/top/pdptw/li-lim-benchmark/

**Reply by cnpryer** [reply]
Awesome. How extensible do you plan to make this?

[Running list](https://github.com/pryborg/vrp/wiki/3PL-Solver-Wishlist) of some of the stuff I'm looking for.

**Reply by reinterpretcat** [reply]
At the moment, scientific formats (solomon, lilim, tsplib) are mostly to support benchmarking which helps to perform analysis with the goal of  improving algorithm. The main focus on json-based format (aka pragmatic) which is designed to support real life VRP variants. Internal API are not yet in the focus, so I can easily break some compatibility between different versions.

Regarding your wish list: 
* Max stops per asset: Maybe maxActivities limit can help here (no the same as stops, but still can help to limit tour size)
* Rate module: costs can be different for different vehicle types, but still "flat" for the whole tour duration/distance. This requirement would require some thinking on implementation design
* DoT - I don't know what it stands for. Breaks are supported, but they are "soft" (with all hard work to assign them)
